command/ca/token.go

```diff
@@ -388,6 +374,67 @@ func newTokenFlow(ctx *cli.Context, subject string, sans []string, caURL, root,
 	return generateToken(subject, sans, kid, issuer, audience, root, notBefore, notAfter, jwk)
 }
 
+func offlineTokenFlow(ctx *cli.Context, subject string, sans []string) (string, error) {
+	caConfig := ctx.String("ca-config")
+	if caConfig == "" {
+		return "", errs.InvalidFlagValue(ctx, "ca-config", "", "")
+	}
+
+	// Using the offline CA
+	if utils.FileExists(caConfig) {
+		offlineCA, err := newOfflineCA(caConfig)
+		if err != nil {
+			return "", err
+		}
+		return offlineCA.GenerateToken(ctx, subject, sans)
+	}
+
+	kid := ctx.String("kid")
+	issuer := ctx.String("issuer")
+	keyFile := ctx.String("key")
+	passwordFile := ctx.String("password-file")
+
+	notBefore, notAfter, err := parseValidity(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// Require kid, issuer and keyFile if ca.json does not exists
+	switch {
+	case len(kid) == 0:
```

Comment:
Can't we assume the kid here since the user passes the keyFile and the default is to just use jose.Thumbprint(jwk)?

Reply:
ok. makes sense